sdk/tools/azure-openrewrite-compiler-maven-plugin/src/main/java/com/azure/openrewrite/goldenimages/GoldenImageCompilerMojo.java

```diff
@@ -0,0 +1,252 @@
+package com.azure.openrewrite.goldenimages;
+
+import org.apache.maven.plugin.AbstractMojo;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugins.annotations.Component;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.project.*;
+import org.apache.maven.artifact.DependencyResolutionRequiredException;
+import org.apache.maven.execution.MavenSession;
+
+import javax.tools.JavaCompiler;
+import javax.tools.ToolProvider;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+@Mojo(name = "compile-golden-image")
+public class GoldenImageCompilerMojo extends AbstractMojo {
+    private static final boolean LIST_DEPENDENCIES = false;
+    private static final boolean LIST_CLASSES = false;
+    private static final boolean LOG_COMPILE_COMMAND = false;
+
+    /**
+     * Comma-separated list of profiles to build.
+     * For example: "v1,v2"
+     */
+    @Parameter(property = "targetProfiles", defaultValue = "v1")
+    private String targetProfiles;
+
+    @Parameter(defaultValue = "${project.basedir}", readonly = true)
+    private File baseDir;
+
+    @Parameter(defaultValue = "${project}", readonly = true, required = true)
+    private MavenProject project;
+
+    @Parameter(defaultValue = "${session}", readonly = true, required = true)
+    private MavenSession session;
+
+    @Component
+    private ProjectBuilder projectBuilder;
+
+    public void execute() throws MojoExecutionException {
+        String testResourcesPath = baseDir.getAbsolutePath() + "/src/test/resources/migrationExamples";
+        File testResourcesDir = new File(testResourcesPath);
+
+        if (!testResourcesDir.exists()) {
+            throw new MojoExecutionException("Test resources directory does not exist: " + testResourcesPath);
+        }
+
+        List<File> testFolders = findTestFolders(testResourcesDir);
+        String[] profiles = targetProfiles.split(",");
+
+        for (File folder : testFolders) {
+            for (String profile : profiles) {
+                File profileDir = new File(folder, profile);
+
+                if (!profileDir.exists()) {
+                    getLog().warn("Skipping " + folder.getAbsolutePath() + " as profile directory " + profile + " does not exist.");
+                    continue;
+                }
+
+                File previousProfileDir = getPreviousProfileDir(folder, profile);
```

Comment:
What does `previous` mean here? Does it mean this is the track 2 directory?

Reply:
Yes